### PR TITLE
Specs for resisting CVE-2011-4815

### PR DESCRIPTION
**Pull request by chrisseaton**
What do people think about creating a new category of specs specifically for named vulnerabilities? We fail these by the way.

### CODE REVIEW

File: spec/ruby/vulnerabilities/cve_2011_4815_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+describe :resists_cve_2011_4815, shared: true do
+  it "reists CVE-2011-4815 by having different hash codes in different processes" do
+    expression = "print (#{@method}).hash"
+    ruby_exe(expression).should_not == ruby_exe(expression)
```
Comment (eregon) [inline spec/ruby/vulnerabilities/cve_2011_4815_spec.rb]: You can save one process invocation by doing an `eval` in this process.

**Comment by eregon** [comment]
For me these specs can be part of ruby/spec since it behavior other Ruby implementations exhibit.
But then of course the specs would be split across multiple files.

There are already slightly related specs like:
```ruby
  it "is different for different instances" do
    Complex(1, 2).hash.should_not == Complex(1, 1).hash
    Complex(2, 1).hash.should_not == Complex(1, 1).hash
    Complex(1, 2).hash.should_not == Complex(2, 1).hash
  end
```

**Comment by chrisseaton** [comment]
> For me these specs can be part of ruby/spec

That's where I've put them. I've just created a new category and if you like the idea I was going to go through all the vulnerabilities listed on the Ruby website and see if it makes sense to write these regression specs for them.

The key thing was referring to the vulnerability by name.

File: spec/ruby/vulnerabilities/cve_2011_4815_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+describe :resists_cve_2011_4815, shared: true do
+  it "reists CVE-2011-4815 by having different hash codes in different processes" do
```
Comment (eregon) [inline spec/ruby/vulnerabilities/cve_2011_4815_spec.rb]: `resists`

**Comment by eregon** [comment]
Oh I see, I thought it was under spec/truffle for some reason.

Yes I think this is a good idea, keeping them together with the issue name/number allows a better context.
Could you do these specs as a PR to [ruby/spec](https://github.com/ruby/spec/) ?

**Comment by chrisseaton** [comment]
Will do. I'll see if I can write some for a couple more of the CVEs at the same time.

**Pull request closed without merging**